### PR TITLE
Template update for nf-core/tools v3.0.2

### PR DESCRIPTION
**Pull request by sateeshperi**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,51 +7,94 @@ on:
...
+        exclude:
+          - isMaster: false
+            profile: "conda"
+          - isMaster: false
+            profile: "singularity"
     steps:
       - name: Check out pipeline code
         uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4
-        with:
-          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (sateeshperi) [reply .github/workflows/ci.yml]: Thank you!

File: .nf-core.yml [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
+  actions_ci: false
   files_exist:
     - lib/nfcore_external_java_deps.jar
     - conf/modules.config
   modules_config: false
+nf_core_version: 3.0.2
+org_path: null
+repository_type: pipeline
+template:
+  author: Phil Ewels
```
Comment (maxulysse) [inline .nf-core.yml:12]: Should we update author at some point, I feel @ewels hasn't been the main dev in this pipeline for a bit

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
-## v2.7.0dev - [2024-10-01]
+## v2.7.0dev - [2024-10-11]
 
 ### Bug fixes & refactoring
 
 - 🔄 Updated template to nf-core/tools v2.12 [#377](https://github.com/nf-core/methylseq/pull/377)
 - 🔄 Updated template to nf-core/tools v2.13 [#380](https://github.com/nf-core/methylseq/pull/380)
 - 🔄 Updated template to nf-core/tools v2.13.1 [#384](https://github.com/nf-core/methylseq/pull/384)
 - 🔄 Updated template to nf-core/tools v2.14.1 [#399](https://github.com/nf-core/methylseq/pull/399)
+- 🔄 Updated template to nf-core/tools v3.0.2 [#426](https://github.com/nf-core/methylseq/pull/426)
```
Comment (maxulysse) [inline CHANGELOG.md:11]: I like the emojis <3

File: CITATIONS.md [hunk 4]
```diff
@@ -16,11 +16,11 @@
 
 - [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
 
-  > Andrews, S. (2010). FastQC: A Quality Control Tool for High Throughput Sequence Data [Online].
+> Andrews, S. (2010). FastQC: A Quality Control Tool for High Throughput Sequence Data [Online].
```
Comment (maxulysse) [inline CITATIONS.md:19]:
```suggestion
  > Andrews, S. (2010). FastQC: A Quality Control Tool for High Throughput Sequence Data [Online].
```

File: CITATIONS.md [hunk 5]
```diff
@@ -16,11 +16,11 @@
...
 
 - [MultiQC](https://pubmed.ncbi.nlm.nih.gov/27312411/)
 
-  > Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.
+> Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.
```
Comment (maxulysse) [inline CITATIONS.md:23]:
```suggestion
  > Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.
```

File: main.nf [hunk 6]
```diff
@@ -30,6 +19,15 @@ include { METHYLSEQ               } from './workflows/methylseq/'
...
 include { PIPELINE_COMPLETION     } from './subworkflows/local/utils_nfcore_methylseq_pipeline'
+include { getGenomeAttribute      } from './subworkflows/local/utils_nfcore_methylseq_pipeline'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    GENOME PARAMETER VALUES
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+params.fasta         = getGenomeAttribute('fasta')
+params.bismark_index = getGenomeAttribute('bismark')
```
Comment (maxulysse) [inline main.nf:30]: I think this needs to be before any calls to PREPARE_GENOME if I remember well

File: main.nf [hunk 7]
```diff
@@ -118,18 +115,6 @@ workflow {
...
-// Get attribute from genome config file e.g. fasta
-//
-def getGenomeAttribute(attribute) {
-    if (params.genomes && params.genome && params.genomes.containsKey(params.genome)) {
-        if (params.genomes[ params.genome ].containsKey(attribute)) {
-            return params.genomes[ params.genome ][ attribute ]
-        }
-    }
-    return null
-}
```
Comment (maxulysse) [inline main.nf:131]: don't you want to keep this here?

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -60,9 +105,3 @@ jobs:
               --changed-since HEAD^ \
               --profile "+${{ matrix.profile }}" \
               --filter pipeline
-
-      - name: Publish Test Report
-        uses: mikepenz/action-junit-report@v3
-        if: always() # always run even if the previous step fails
-        with:
-          report_paths: test.xml
```
[the remark this answers is not in the document]
Comment (sateeshperi) [reply .github/workflows/ci.yml:68]: not if you all are also not using it

File: main.nf (hunk 7 above)
Comment (sateeshperi) [reply main.nf:131]: actually it works now from the function being included explicitly at top
`include { getGenomeAttribute      } from './subworkflows/local/utils_nfcore_methylseq_pipeline'`

File: .github/workflows/ci.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (sateeshperi) [reply .github/workflows/ci.yml:68]: u got rid of this publish ?

File: main.nf (hunk 6 above)
Comment (sateeshperi) [reply main.nf:30]: it is before any calls to PREPARE_GENOME

File: .github/workflows/ci.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (sateeshperi) [reply .github/workflows/ci.yml:68]: yes will put back in

File: .nf-core.yml (hunk 2 above)
Comment (sateeshperi) [reply .nf-core.yml:12]: I will check with him on methylseq ch and resolve this in a future PR

**Pull request merged**